graph/resolver/schema.resolvers.go

```diff
@@ -667,6 +667,100 @@ func (r *mutationResolver) ShuffleDiscussion(ctx context.Context, discussionID s
 	return r.DAOManager.GetDiscussionByID(ctx, discussionID)
 }
 
+func (r *mutationResolver) MuteParticipants(ctx context.Context, discussionID string, participantIDs []string, mutedForSeconds int) ([]*model.Participant, error) {
+	if mutedForSeconds < 0 || mutedForSeconds > 86400 {
+		return nil, fmt.Errorf("mutedForSeconds value is invalid")
+	}
+
+	authedUser := auth.GetAuthedUser(ctx)
+	if authedUser == nil {
+		return nil, fmt.Errorf("Need auth")
+	}
+
+	/* Only moderators can use this mutation */
+	modCheck, err := r.DAOManager.CheckIfModerator(ctx, authedUser.UserID)
+	if err != nil || !modCheck {
+		return nil, fmt.Errorf("unauthorized")
+	}
+
+	/* Get discussion participants */
+	participants, err := r.DAOManager.GetParticipantsByDiscussionID(ctx, discussionID)
+	if err != nil {
+		return nil, err
+	} else if participants == nil {
+		return nil, fmt.Errorf("Error fetching participants with discussionID (%s)", discussionID)
+	} else if len(participants) == 0 {
+		return []*model.Participant{}, nil
+	}
+
+	/* Check participants validity and retrieve the ones we need to modify */
+	var participantsToEdit []*model.Participant
```

Comment:
We should probably move this logic into the `backend` package so it's tested. It's not really resolver logic.

Reply:
Yeah you are right. I'm gonna open a pr for this.